code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -56,19 +60,23 @@
 	var/delivery_icon = "deliverycloset" //which icon to use when packagewrapped. null to be unwrappable.
 	var/anchorable = TRUE
 	var/icon_welded = "welded"
-	/// How close being inside of the thing provides complete pressure safety. Must be between 0 and 1!
-	contents_pressure_protection = 0
-	/// How insulated the thing is, for the purposes of calculating body temperature. Must be between 0 and 1!
-	contents_thermal_insulation = 0
 	/// Whether a skittish person can dive inside this closet. Disable if opening the closet causes "bad things" to happen or that it leads to a logical inconsistency.
 	var/divable = TRUE
 	/// true whenever someone with the strong pull component (or magnet modsuit module) is dragging this, preventing opening
 	var/strong_grab = FALSE
-	///electronics for access
-	var/obj/item/electronics/airlock/electronics
+	/// secure locker or not, also used if overriding a non-secure locker with a secure door overlay to add fancy lights
+	var/secure = FALSE
 	var/can_install_electronics = TRUE
 
 	var/contents_initialized = FALSE
+	/// is this closet locked by an exclusive id, i.e. your own personal locker
+	var/datum/weakref/id_card = null
+	/// should we prevent furthur access change
+	var/access_locked = FALSE
+	/// is the card reader installed in this machine
+	var/card_reader_installed = FALSE
+	/// access types for card reader
+	var/list/access_choices = list("Personal", "Departmental", "None")
```

Comment:
Efficient

Reply:
We can't do that actually because this list is overriden by `/obj/structure/closet/secure_closet/personal` so players can't set departmental/ free access for closets. Although i can assign this variable with a static list, I'll see what i can do

Reply:
Maybe add a `can_override_access` boolean var that you need to check before using the list?  There is a massive amount of lockers and crates that spawn so this optimization would be a good idea.